packages/open_aea/connections/p2p_libp2p/libp2p_node/utils/utils.go

```diff
@@ -596,6 +629,21 @@ func IDFromFetchAIPublicKeyUncompressed(publicKey string) (peer.ID, error) {
 	return multihash, nil
 }
 
+// IDFromEthereumPublicKey Get PeerID (multihash) from ethereum public key
+func IDFromEthereumPublicKey(publicKey string) (peer.ID, error) {
+	// Uncompressed
+	pubKey, err := BTCPubKeyFromUncompressedHex(publicKey[2:]) // slice off "0x"
```

Comment:
that is an excellent question to which I do not know the answer. Perhaps because 3 crypto libraries were not deemed sufficient? I simply implemented it the same way the fetchAI keys are converted to an peerID with only the minimal amount of changes required for it pass the tests. Let's keep this comment alive for a refactoring session once all is done and functional.

Reply:
it's a good question, I don't know. What I do know tho, is that ethereum uses the same elliptic curve as bitcoin (secp256k1) so deriving the public key is identical